### ISSUE TITLE
kubectl hns tree command very slow

### ISSUE THREAD

**Issue by markbaylis** [kubernetes-sigs/hierarchical-namespaces#223]
Hi,

This is more cosmetic than anything else, but we have observed the 'kubectl hns tree' command is very slow to draw out the tree.

Is there any plans to improve this?

Thanks
Mark

**Comment by adrianludwin** [comment]
I haven't observed this, could it be that your apiserver is throttling
replies? I'd generally expect one call to the apiserver per namespace,
*maybe* two. How big is the tree you're trying to draw, and how long does
it take? Is it faster if you just do a subset of the tree?

I suspect we could improve performance by using List calls instead to
retrieve lots of information at the same time, though this wouldn't work if
the caller didn't have read permissions to the entire cluster. Otherwise,
we'd have to precompute data somewhere on the server and I'm not too
excited about that either. I suppose HNC could support some additional
endpoint (e.g. as an apiserver extension) but that seems like a lot of
trouble as well.

In short, there aren't any plans to improve this and I'm not really sure
where I'd start. That's assuming we don't find some clear bug, of course,
which is certainly possible.

**Comment by adrianludwin** [comment]
See also #226
